sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -1220,12 +1281,40 @@ public CaseDataDto saveCase(CaseDataDto dto, boolean handleChanges) throws Valid
 
 		caseService.ensurePersisted(caze);
 		if (handleChanges) {
+			updateCaseVisitAssociations(existingCaseDto, caze);
+			caseService.updateFollowUpUntilAndStatus(caze);
+
 			onCaseChanged(existingCaseDto, caze);
 		}
 
 		return convertToDto(caze);
 	}
 
+	private void updateCaseVisitAssociations(CaseDataDto existingCase, Case caze) {
```

Comment:
When a contact becomes a case, should visits to the contact be seen on the case?

@carolinverset @OlivierLeuci your opinion?

Reply:
Yes, if it's possible, i think it would be interesting to keep the history of follow-up visits of the contact turned case

Reply:
I agree that we can review the data model, a visit can only be associated with one case.
However, I am not sure I can see what parts of the code can be simplified.
Shouldn't the logic of associating visits to a case based on dates be kept so that it is consistent with the input control on the visit form?
It seems to me that changing the end date of follow-up of a case can lead to the association (or the disassociation) of visits to this case.

Reply:
I think another way of viewing such a history would be to have a link back to the source contact at the case, which would then allow users to have a look at the visits associated with the contact before it was turned into a case. 

@xavier-calland It wouldn't because we would have a 1 (case) to n (visits) relation, i.e. a visit would be strictly assigned to a single case and then be displayed in the follow-up list of that case irrespective of the follow-up end date. We don't need the whole logic we're using for contacts because, as I've written, one person can have multiple contacts at the same time, but only one case (of the same disease).